src/main/java/com/minecolonies/colony/buildings/BuildingGuardTower.java

```diff
@@ -167,11 +370,152 @@ public int getBonusVision()
         return MAX_VISION_BONUS_MULTIPLIER * VISION_BONUS;
     }
 
+    @Override
+    public void readFromNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.readFromNBT(compound);
+        task = Task.values()[compound.getInteger(TAG_TASK)];
+        int jobId = compound.getInteger(TAG_JOB);
+        job = jobId == -1 ? null : GuardJob.values()[jobId];
+        assignManually = compound.getBoolean(TAG_ASSIGN);
+        retrieveOnLowHealth = compound.getBoolean(TAG_RETRIEVE);
+        patrolManually = compound.getBoolean(TAG_PATROL);
+
+        final NBTTagList wayPointTagList = compound.getTagList(TAG_PATROL_TARGETS, Constants.NBT.TAG_COMPOUND);
+        for (int i = 0; i < wayPointTagList.tagCount(); ++i)
+        {
+            NBTTagCompound blockAtPos = wayPointTagList.getCompoundTagAt(i);
+            final BlockPos pos = BlockPosUtil.readFromNBT(blockAtPos, TAG_TARGET);
+            patrolTargets.add(pos);
+        }
+
+        guardPos = BlockPosUtil.readFromNBT(compound, TAG_GUARD);
+    }
+
+    @Override
+    public void writeToNBT(@NotNull final NBTTagCompound compound)
+    {
+        super.writeToNBT(compound);
+        compound.setInteger(TAG_TASK, task.ordinal());
+        compound.setInteger(TAG_JOB, job == null ? -1 : job.ordinal());
+        compound.setBoolean(TAG_ASSIGN, assignManually);
+        compound.setBoolean(TAG_RETRIEVE, retrieveOnLowHealth);
+        compound.setBoolean(TAG_PATROL, patrolManually);
+
+        @NotNull final NBTTagList wayPointTagList = new NBTTagList();
+        for (@NotNull final BlockPos pos: patrolTargets)
+        {
+            @NotNull final NBTTagCompound wayPointCompound = new NBTTagCompound();
+            BlockPosUtil.writeToNBT(wayPointCompound, TAG_TARGET, pos);
+
+
+            wayPointTagList.appendTag(wayPointCompound);
+        }
+        compound.setTag(TAG_PATROL_TARGETS, wayPointTagList);
+
+        BlockPosUtil.writeToNBT(compound, TAG_GUARD, guardPos);
+    }
+
+    @Override
+    public void serializeToView(@NotNull final ByteBuf buf)
+    {
+        super.serializeToView(buf);
+        buf.writeBoolean(assignManually);
+        buf.writeBoolean(retrieveOnLowHealth);
+        buf.writeBoolean(patrolManually);
+        buf.writeInt(task.ordinal());
+        buf.writeInt(job == null ? -1 : job.ordinal());
+        buf.writeInt(patrolTargets.size());
+
+        for(BlockPos pos: patrolTargets)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'pos' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)